diff --git syntaxes/01_Packages/Java/JavaDoc.sublime-syntax syntaxes/01_Packages/Java/JavaDoc.sublime-syntax
index 422a6a9..40a741e 100644
--- syntaxes/01_Packages/Java/JavaDoc.sublime-syntax
+++ syntaxes/01_Packages/Java/JavaDoc.sublime-syntax
@@ -13,7 +13,7 @@ variables:
 contexts:
   prototype:
     # https://docs.oracle.com/javase/7/docs/technotes/tools/windows/javadoc.html#leadingasterisks
-    - match: ^\s*(\*)\s*(?!\s*@)
+    - match: ^\s*(\*)(?!/)\s*(?!\s*@)
       captures:
         1: punctuation.definition.comment.javadoc
 
