diff --git themes/sublime-monokai-extended/Monokai\ Extended.tmTheme themes/sublime-monokai-extended/Monokai\ Extended.tmTheme
index 9c2aa3e..180cbbf 100644
--- themes/sublime-monokai-extended/Monokai Extended.tmTheme	
+++ themes/sublime-monokai-extended/Monokai Extended.tmTheme	
@@ -810,11 +810,11 @@
       <key>name</key>
       <string>JSON String</string>
       <key>scope</key>
-      <string>meta.structure.dictionary.json string.quoted.double.json</string>
+      <string>meta.mapping.key.json string.quoted.double.json, punctuation.separator.sequence.csv</string>
       <key>settings</key>
       <dict>
         <key>foreground</key>
-        <string>#cfcfc2</string>
+        <string>#fd971f</string>
       </dict>
     </dict>
     <dict>
@@ -1027,7 +1027,7 @@
       <key>name</key>
       <string>Invalid</string>
       <key>scope</key>
-      <string>invalid</string>
+      <string>invalid, meta.annotation.error-line</string>
       <key>settings</key>
       <dict>
         <key>background</key>
@@ -1038,11 +1038,22 @@
         <string>#f8f8f0</string>
       </dict>
     </dict>
+    <dict>
+      <key>name</key>
+      <string>Error</string>
+      <key>scope</key>
+      <string>markup.error</string>
+      <key>settings</key>
+      <dict>
+        <key>foreground</key>
+        <string>#dd2020</string>
+      </dict>
+    </dict>
     <dict>
       <key>name</key>
       <string>Invalid deprecated</string>
       <key>scope</key>
-      <string>invalid.deprecated</string>
+      <string>invalid.deprecated, markup.warning</string>
       <key>settings</key>
       <dict>
         <key>background</key>
