--- themes/sublime-monokai-extended/Monokai Extended.tmTheme	
+++ themes/sublime-monokai-extended/Monokai Extended.tmTheme	
@@ -808,13 +808,24 @@
     </dict>
     <dict>
       <key>name</key>
+      <string>Markdown Table Delimiters</string>
+      <key>scope</key>
+      <string>punctuation.separator.table-cell, punctuation.section.table-header</string>
+      <key>settings</key>
+      <dict>
+        <key>foreground</key>
+        <string>#66d9ef</string>
+      </dict>
+    </dict>
+    <dict>
+      <key>name</key>
       <string>JSON String</string>
       <key>scope</key>
-      <string>meta.structure.dictionary.json string.quoted.double.json</string>
-      <key>settings</key>
-      <dict>
-        <key>foreground</key>
-        <string>#cfcfc2</string>
+      <string>meta.mapping.key.json string.quoted.double.json, punctuation.separator.sequence.csv</string>
+      <key>settings</key>
+      <dict>
+        <key>foreground</key>
+        <string>#fd971f</string>
       </dict>
     </dict>
     <dict>
@@ -1027,7 +1038,7 @@
       <key>name</key>
       <string>Invalid</string>
       <key>scope</key>
-      <string>invalid</string>
+      <string>invalid, meta.annotation.error-line</string>
       <key>settings</key>
       <dict>
         <key>background</key>
@@ -1040,9 +1051,20 @@
     </dict>
     <dict>
       <key>name</key>
+      <string>Error</string>
+      <key>scope</key>
+      <string>markup.error</string>
+      <key>settings</key>
+      <dict>
+        <key>foreground</key>
+        <string>#dd2020</string>
+      </dict>
+    </dict>
+    <dict>
+      <key>name</key>
       <string>Invalid deprecated</string>
       <key>scope</key>
-      <string>invalid.deprecated</string>
+      <string>invalid.deprecated, markup.warning</string>
       <key>settings</key>
       <dict>
         <key>background</key>
