--- syntaxes/01_Packages/ShellScript/Bash.sublime-syntax
+++ syntaxes/01_Packages/ShellScript/Bash.sublime-syntax
@@ -33,2 +33,2 @@
-  - .ebuild
-  - .eclass
+  - ebuild
+  - eclass
@@ -38 +38 @@
-    ^\#! .* \b(bash|zsh|sh|tcsh|ash)\b
+    ^\#! .* \b(bash|zsh|sh|tcsh|ash|dash|ksh)\b
@@ -159,0 +160,3 @@
+    - match: \bselect{{keyword_break}}
+      scope: keyword.control.loop.select.shell
+      set: [cmd-post, select-args]
@@ -259,0 +263,8 @@
+  select-args:
+    - match: ""
+      set:
+        - meta_scope: meta.group.select.shell
+        - include: cmd-args-boilerplate
+        - match: \bin{{keyword_break}}
+          scope: keyword.control.in.shell
+
@@ -344 +355 @@
-      - vardef-maybe-more
+      - vardef-alias-maybe-more
@@ -416,0 +428,8 @@
+  vardef-alias-maybe-more:
+    - meta_include_prototype: false
+    - match: (?=`|\s*#)
+      pop: true
+    - include: cmd-args-boilerplate
+    - match: (?=\S)
+      push: [vardef-value, vardef-assign, vardef-alias-name]
+
@@ -418 +437,5 @@
-    - match: \s*((-)p)
+    - match: \s*(--)(?=\s|$)
+      captures:
+        1: keyword.operator.end-of-options.shell
+      pop: true
+    - match: \s*((-)p)(?=\s|$)
@@ -432 +455 @@
-        - match: (?={{varassign}}|\s)|$
+        - match: (?==|\s|[;&`]|{{metachar}})|$
